### DISCUSSION TITLE
How to change the worker script to self hosted version with file download?

### DISCUSSION THREAD

**Discussion by Schokolader** [plausible/analytics#2461] [Self-Hosted Support]
Hello, 

how would you change the worker script: 
[](https://plausible.io/docs/proxy/guides/cloudflare )
to a self hosted version with file download? 

Somehow I don't get this running.

**Comment by Schokolader** [comment]
Ok got it running. 

Now it is behind a Cloudflare worker. But the only thing is that with file downloads, now only the direct links to the files are tracked. 
If you use pretty links, the download will not be tracked. 
Previously just with the normal script.file-downloads.js  version it worked also for pretty links.

**Reply by abdessalaam** [reply]
How did you make file-downloads.js work with self-hosted version? I have a MIME type mismatch error.
(outbound-links script works fine)

**Reply by Schokolader** [reply]
Have you installed the last RC version? You need 1.5.0-rc.0 or v1.5.0-rc.1

**Comment by abdessalaam** [comment]
Thanks, I have the latest stable version. This explains it!
